lib/handler.js

```diff
@@ -73,52 +73,61 @@ class Handler {
   static async handle (context, pullRequest) {
     const checkRunResult = await checks.create(context, 'Mergeable')
     // let the user know that we are validating if PR is mergeable
-
-    var config = await Configuration.instanceWithContext(context)
-
-    let settings = config.settings.mergeable.pull_requests || config.settings.mergeable
-    let validators = []
-    let excludes = (settings.exclude)
-      ? settings.exclude.split(',').map(val => val.trim()) : []
-    let includes = [ 'approvals', 'title', 'label', 'milestone', 'description', 'projects', 'assignee', 'dependent' ]
-      .filter(validator => excludes.indexOf(validator) === -1)
-
-    includes.forEach(validator => {
-      validators.push(require(`../lib/${validator}`)(pullRequest, context, settings))
-    })
-
-    console.info(config)
-    return Promise.all(validators).then(results => {
-      let failures = results.filter(validated => !validated.mergeable)
-
-      let status, description
-      if (failures.length === 0) {
-        status = 'success'
-        description = 'Okay to merge.'
-      } else {
-        status = 'failure'
-        description = `## Mergeable has found the following failed checks`
-        for (let validated of failures) {
-          description += `\n - ${validated.description}`
+    try {
+      var config = await Configuration.instanceWithContext(context)
```

Comment:
IMO the try catch should be around this statement where it is an actual error from loading / processing the configuration.

Reply:
i thought we want to do a general try and catch for all unknown errors

Reply:
The original catch was on `Promise.all` and would have caught any promise rejections in this entire function.

The only other place where an error could occur is during the loading of Configuration. A try catch around that with the very specific message telling the user to fix their configuration will suffice. It is actionable. The promise rejections from the validators are unlikely to be actionable. It deserves it's own message and an error log since it's most likely to be a bug.